packages/backend/src/payment-method/local/service.ts

```diff
@@ -0,0 +1,294 @@
+  isConvertError,
+  RateConvertOpts,
+  RatesService
+} from '../../rates/service'
+import { IAppConfig } from '../../config/app'
+import {
+  PaymentMethodHandlerError,
+  PaymentMethodHandlerErrorCode
+} from '../handler/errors'
+import {
+  AccountingService,
+  LiquidityAccountType,
+  TransferOptions,
+  TransferType
+} from '../../accounting/service'
+import {
+  AccountAlreadyExistsError,
+  errorToMessage,
+  isTransferError,
+  TransferError
+} from '../../accounting/errors'
+import { IncomingPaymentService } from '../../open_payments/payment/incoming/service'
+import { IncomingPaymentState } from '../../open_payments/payment/incoming/model'
+
+export interface LocalPaymentService extends PaymentMethodService {}
+
+export interface ServiceDependencies extends BaseService {
+  config: IAppConfig
+  ratesService: RatesService
+  accountingService: AccountingService
+  incomingPaymentService: IncomingPaymentService
+}
+
+export async function createLocalPaymentService(
+  deps_: ServiceDependencies
+): Promise<LocalPaymentService> {
+  const deps: ServiceDependencies = {
+    ...deps_,
+    logger: deps_.logger.child({ service: 'LocalPaymentService' })
+  }
+
+  return {
+    getQuote: (quoteOptions) => getQuote(deps, quoteOptions),
+    pay: (payOptions) => pay(deps, payOptions)
+  }
+}
+
+async function getQuote(
+  deps: ServiceDependencies,
+  options: StartQuoteOptions
+): Promise<PaymentQuote> {
+  const { receiver, debitAmount, receiveAmount } = options
+
+  let debitAmountValue: bigint
+  let receiveAmountValue: bigint
+
+  const convert = async (opts: RateConvertOpts) => {
+    let converted: bigint | ConvertError
+    try {
+      converted = await deps.ratesService.convert(opts)
+    } catch (err) {
+      deps.logger.error(
+        { receiver, debitAmount, receiveAmount, err },
+        'Unknown error while attempting to convert rates'
+      )
+      throw new PaymentMethodHandlerError(
+        'Received error during local quoting',
+        {
+          description: 'Unknown error while attempting to convert rates',
+          retryable: false
+        }
+      )
+    }
+    return converted
+  }
+
+  if (debitAmount) {
+    debitAmountValue = debitAmount.value
+    const converted = await convert({
+      sourceAmount: debitAmountValue,
+      sourceAsset: {
+        code: debitAmount.assetCode,
+        scale: debitAmount.assetScale
+      },
+      destinationAsset: {
+        code: receiver.assetCode,
+        scale: receiver.assetScale
+      }
+    })
+    if (isConvertError(converted)) {
+      throw new PaymentMethodHandlerError(
+        'Received error during local quoting',
+        {
+          description: 'Failed to convert debitAmount to receive amount',
+          retryable: false
+        }
+      )
+    }
+    receiveAmountValue = converted
+  } else if (receiveAmount) {
+    receiveAmountValue = receiveAmount.value
+    const converted = await convert({
+      sourceAmount: receiveAmountValue,
```

Comment:
I hardcoded in the rates because I'm not sure where to get the sending currency (USD), since there is no `debitAmount` here. Also, I'm not actually sure how to trigger this block as opposed to `receiver.incomingAmount` further down. That's what the fixed send p2p example triggers. Although I think your comment applies to both places so that detail probably doesnt matter.

Anyways, just wanted to validate this. With your calculation patched in and doing a fixed send local payment, I see an outgoing payment with these amounts:

```json
      "receiveAmount": {
        "assetCode": "EUR",
        "assetScale": 2,
        "value": "1000"
      },
      "debitAmount": {
        "assetCode": "USD",
        "assetScale": 2,
        "value": "1221"
      },
      "sentAmount": {
        "assetCode": "USD",
        "assetScale": 2,
        "value": "1099"
      },
```

The way it is now, the `receiveAmount` is the same and `debitAmount` and `sentAmount` are 1 higher:

```json
      "receiveAmount": {
        "assetCode": "EUR",
        "assetScale": 2,
        "value": "1000"
      },
      "debitAmount": {
        "assetCode": "USD",
        "assetScale": 2,
        "value": "1222"
      },
      "sentAmount": {
        "assetCode": "USD",
        "assetScale": 2,
        "value": "1100"
      },
```

For the same payment over ILP I see the same `1099` `sentAmount` from your revised calculation but a different `debitAmount`. Not sure if slippage would account for this (in which case there is no problem) or something else.

```json
      "receiveAmount": {
        "assetCode": "EUR",
        "assetScale": 2,
        "value": "1000"
      },
      "debitAmount": {
        "assetCode": "USD",
        "assetScale": 2,
        "value": "1234"
      },
      "sentAmount": {
        "assetCode": "USD",
        "assetScale": 2,
        "value": "1099"
      },
```

Reply:
Also, the estimated exchange rate can be the exchange rate that's received directly from the rates service, instead of having to do `estimatedExchangeRate: Number(receiveAmountValue) / Number(debitAmountValue)`

Reply:
🤦 Of course, the `destinationAsset` passed in to the rates call.